.github/workflows/release.yaml

```diff
@@ -0,0 +1,177 @@
+            const owner = context.repo.owner
+            const repo = context.repo.repo
+
+            const attached_labels = (await github.rest.issues.listLabelsOnIssue({ owner, repo, issue_number })).data
+
+            const is_release_label_attached = attached_labels.find(label => label.name == 'release') !== undefined
+            core.setOutput('is_release_label_attached', is_release_label_attached)
+  release:
+    runs-on: ubuntu-latest
+    needs: check_label
+    if: needs.check_label.outputs.is_release_label_attached == 'true'
+    env:
+      EARTHLY_TOKEN: "${{ secrets.EARTHLY_TOKEN }}"
+      CARGO_REGISTRY_TOKEN: "${{ secrets.CARGO_REGISTRY_TOKEN }}"
+      FORCE_COLOR: 1
+    steps:
+      - uses: earthly/actions-setup@v1
+        with:
+          version: v0.8.3
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+      - uses: jwalton/gh-find-current-pr@v1
+        id: find_pr
+        with:
+          state: all
+      - run: npm install toml globby axios
+      - uses: actions/github-script@v7
+        with:
+          script: |
+            const is_merge_event = ${{ github.event_name == 'push' }}
+
+            const { globby } = await import("${{ github.workspace }}/node_modules/globby/index.js");
+            const toml = require('toml');
+            const fs = require('fs');
+            const axios = require('axios');
+            const { spawn } = require('child_process');
+
+            async function get_crate_details(name) {
+              try {
+                return (await axios(`https://crates.io/api/v1/crates/${name}`)).data
+              } catch(error) {
+                console.error("Error fetching crate details:", error)
+              }
+            }
+
+            function run_command_stream(command_arr) {
+              const [command, ...args] = command_arr
+
+              const child = spawn(command, args)
+              child.stdout.on("data", data => console.log(Buffer.from(data).toString()))
+              child.stderr.on("data", data => console.error(Buffer.from(data).toString()))
+              child.on("close", code => {
+                console.log(command, "exited with code", code)
+                if (code != 0) {
+                  core.setFailed(`Command ${command_str} exited with code ${code}`)
+                }
+              })
+            }
+
+            async function create_or_update_comment(identifier, comment_body) {
+              const comment_header = `<!-- ${identifier} -->\n`
+              const comment = comment_header + comment_body
+
+              const issue_number = ${{ steps.find_pr.outputs.pr }}
+              const owner = context.repo.owner
+              const repo = context.repo.repo
+
+              const existing_comments = (await github.rest.issues.listComments({
+                  issue_number, owner, repo
+              })).data
```

Comment:
FYI you could use peter-evans/find-comment@v3 and peter-evans/create-or-update-comment@v4 to replace a bunch of this.

Reply:
For logic that is as simple as this one, how should we choose whether to use a library or implement it ourselves?

Reply:
I generally lean towards using exising/debugged libraries, at least if they seem reasonably widely used.

Reply:
Updated to use the pre-built actions in 3d26d56